src/main/kotlin/org/rust/lang/core/resolve/LangItems.kt

```diff
@@ -35,13 +36,39 @@ fun findIteratorItemType(project: Project, ty: RustType): RustType {
     return rawType.substitute(typeParameterMap)
 }
 
+fun findIndexOutputType(project: Project, containerType: RustType, indexExprType: RustType): RustType {
+    val impls = RsImplIndex.findImpls(project, containerType)
+    for (impl in impls) {
+        val trait = impl.traitRef?.resolveToTrait ?: continue
+        if (trait.isIndex) {
+            // TODO: get index type from impl declaration
+            val indexType = impl.functionList
```

Comment:
Sorry, I for some reason thought that this is not for the `Index` type, but for the output type. 

Yeah, for the index type I think it's easiest just to look at the function signature, but, as per comment below, we should only use it if we find more than one `Index` for one type and need to select between them.